pkg/common/config/eventingkafkaconfig.go

```diff
@@ -58,17 +60,43 @@ type EKCloudEventConfig struct {
 // EKKafkaConfig contains items relevant to Kafka specifically, and the Sarama logging flag
 type EKKafkaConfig struct {
 	Brokers             string             `json:"brokers,omitempty"`
-	EnableSaramaLogging bool               `json:"enableSaramaLogging,omitempty"`
-	Topic               EKKafkaTopicConfig `json:"topic,omitempty"`
-	AdminType           string             `json:"adminType,omitempty"`
 	AuthSecretName      string             `json:"authSecretName,omitempty"`
 	AuthSecretNamespace string             `json:"authSecretNamespace,omitempty"`
+	Topic               EKKafkaTopicConfig `json:"topic,omitempty"`
+}
+
+// EKDistributedConfig holds fields needed only by the Distributed Channel component
+type EKDistributedConfig struct {
+	Receiver  EKReceiverConfig `json:"receiver,omitempty"`
+	AdminType string           `json:"adminType,omitempty"`
+}
+
+// EKConsolidatedConfig is reserved for configuration fields needed only by the Consolidated Channel component
+type EKConsolidatedConfig struct {
+}
+
+// EKSourceConfig is reserved for configuration fields needed by the Kafka Source component
+type EKSourceConfig struct {
+}
+
+// EKChannelConfig contains items relevant to the eventing-kafka channels (distributed and consolidated)
+type EKChannelConfig struct {
+	Dispatcher   EKDispatcherConfig   `json:"dispatcher,omitempty"`
+	Distributed  EKDistributedConfig  `json:"distributed,omitempty"`
```

Comment:
just wondering if the ConfigMap would "read" better (more consistent) if the `Dispatcher` was under Distributed AND Consolidated.  I realize it is "common" to both Channels but a third channel could come along with a different architecture?  It also aligns nicely next to the "Receiver" in distributed case.  Users would only ever have one type of ConfigMap (consolidated / distributed) so it wouldn't appear in there twice.   Anyway - not a big deal, just a minor organziation preference - would be good to hear from others as well ; )

Reply:
I tried it out and it does look less wonky that way.  New configmap format:
```
  eventing-kafka: |
    ...
    channel:
      distributed:
        adminType: kafka
        dispatcher:
          cpuRequest: 100m
          memoryRequest: 50Mi
        receiver:
          cpuRequest: 100m
          memoryRequest: 50Mi
```
or
```
  eventing-kafka: |
    ...
    channel:
      consolidated:
        dispatcher:
          cpuRequest: 100m
          memoryRequest: 50Mi
```
And as an extra bonus, the EKConsolidatedConfig struct is no longer "reserved for future use"